source/fab/tools/tool_box.py

```diff
@@ -46,19 +46,29 @@ def add_tool(self, tool: Tool,
                           f"'{tool}'.")
         self._all_tools[tool.category] = tool
 
-    def get_tool(self, category: Category) -> Tool:
+    def get_tool(self, category: Category, mpi: Optional[bool] = None) -> Tool:
         '''Returns the tool for the specified category.
 
         :param category: the name of the category in which to look
             for the tool.
+        :param mpi: if no compiler or linker is specified when requesting one,
+            use the MPI setting to find an appropriate default.
 
         :raises KeyError: if the category is not known.
         '''
 
         if category in self._all_tools:
+            # TODO: Should we test if the compiler has MPI support if
+            # required? The original LFRic setup compiled files without
+            # MPI support (and used an mpi wrapper at link time), so for
+            # now we don't raise an exception here to ease porting - but
+            # we probably should raise one tbh.
```

Comment:
Same as above. I believe nowadays we can assume that compiler support openmp

Reply:
We could but assumptions tend to come back to haunt people.

If the amount of work involved in checking for compiler support is small then we should just go ahead and do it. If it's more extensive then we need a design which will support it in the future and some sign-posts to note where it would go.

Reply:
The amount of work is small (testing is probably more work, since we get more exceptions). I've added this all (but tbh, I think that could have been done the moment we need to support a compiler that does not support OpenMP). Anyway, done.